libs/ui-components/src/components/EnrollmentRequest/EnrollmentRequestList.tsx

```diff
@@ -79,8 +76,14 @@ const EnrollmentRequestList = ({ refetchDevices, isStandalone }: EnrollmentReque
     },
   });
 
-  // In non-standalone mode, hide the entire component when there are no pending enrollments
-  if (!isStandalone && pendingEnrollments.length === 0) {
+  React.useEffect(() => {
+    if (!search && !isLoading) {
+      setIsLastUnfilteredListEmpty(itemCount === 0);
+    }
+  }, [itemCount, search, isLoading]);
+
+  // In non-standalone mode, hide the entire component when the search result is empty (and not due to filtering)
+  if (!isStandalone && isLastUnfilteredListEmpty) {
```

Comment:
Why are you storing it in state ? Seems like we can just calculate it render loop
```suggestion
  const isLastUnfilteredListEmpty = !search && !isLoading && itemCount === 0

  // In non-standalone mode, hide the entire component when the search result is empty (and not due to filtering)
  if (!isStandalone && isLastUnfilteredListEmpty) {
```

Reply:
You are right, I'll use your patch.